### PR TITLE
fix: nil pointer error

### PR DESCRIPTION
**Pull request by vLia**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/tcl/apitcl/v1/testworkflowmetrics.go [hunk 1]
```diff
@@ -178,10 +182,12 @@ func (s *apiTCL) sendRunWorkflowTelemetry(ctx context.Context, workflow *testwor
 	}
 
 	var dataSource string
-	if len(workflow.Spec.Content.Files) != 0 {
-		dataSource = "files"
-	} else if workflow.Spec.Content.Git != nil {
-		dataSource = "git"
+	if workflow.Spec.Content != nil {
```
Comment (olensmar) [inline pkg/tcl/apitcl/v1/testworkflowmetrics.go:185]: looks like this is the same code as above - could this be extracted into a shared method?

Comment (vLia) [reply pkg/tcl/apitcl/v1/testworkflowmetrics.go:180]: good catch, just pushed an update

File: pkg/tcl/apitcl/v1/testworkflowmetrics.go [hunk 2]
```diff
@@ -24,56 +24,19 @@ func (s *apiTCL) sendCreateWorkflowTelemetry(ctx context.Context, workflow *test
...
 		},
 		WorkflowParams: telemetry.WorkflowParams{
 			TestWorkflowSteps:          int32(len(workflow.Spec.Steps)),
 			TestWorkflowTemplateUsed:   len(workflow.Spec.Use) != 0,
-			TestWorkflowImage:          image,
-			TestWorkflowArtifactUsed:   hasArtifacts,
-			TestWorkflowKubeshopGitURI: isKubeshopGitURI,
+			TestWorkflowImage:          getImage(workflow.Spec.Container),
+			TestWorkflowArtifactUsed:   hasArtifacts(workflow.Spec.Steps),
+			TestWorkflowKubeshopGitURI: isKubeshopGitURI(workflow.Spec.Content),
```
Comment (rangoo94) [inline pkg/tcl/apitcl/v1/testworkflowmetrics.go:39]: Content may be deeper as well in steps

File: pkg/tcl/apitcl/v1/testworkflowmetrics.go [hunk 3]
```diff
@@ -24,56 +24,19 @@ func (s *apiTCL) sendCreateWorkflowTelemetry(ctx context.Context, workflow *test
...
 			AppVersion: version.Version,
-			DataSource: dataSource,
-			Host:       host,
-			ClusterID:  clusterID,
+			DataSource: getDataSource(workflow.Spec.Content),
+			Host:       getHostname(),
+			ClusterID:  s.getClusterID(ctx),
 		},
 		WorkflowParams: telemetry.WorkflowParams{
 			TestWorkflowSteps:          int32(len(workflow.Spec.Steps)),
```
Comment (rangoo94) [inline pkg/tcl/apitcl/v1/testworkflowmetrics.go:35]: There are both `spec.steps`, `spec.setup` and `spec.after`

**Review by rangoo94 — COMMENTED** [review]
Some comments to the telemetry problems, although rather unrelated to this bigger problem

**Review by rangoo94 — APPROVED** [review]

**Review by olensmar — APPROVED** [review]

**Comment by vLia** [comment]
Will create a new PR with cherry-pick from the commit in main

**Pull request closed without merging**